File: C#/ASCII Snake/ASCII Snake/Vector2Int.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ASCII_Snake
{
    public struct Vector2Int
    {
        int _x;
        int _y;

        public int GetX => _x;
        public int GetY => _y;

        public Vector2Int(int x, int y)
        {
            _x = x;
            _y = y;
        }

        public static Vector2Int Zero()
        {
            return new Vector2Int(0, 0);
        }

        // Operators

        public static Vector2Int operator + (Vector2Int a, Vector2Int b)
        {
            return new Vector2Int(a.GetX + b.GetX, a.GetY + b.GetY);
        }

        public static Vector2Int operator - (Vector2Int a, Vector2Int b)
        {
            return new Vector2Int(a.GetX - b.GetX, a.GetY - b.GetY);
        }

        public static bool operator == (Vector2Int a, Vector2Int b)
        {
            if (a.GetX == b.GetX && a.GetY == b.GetY)
                return true;

            return false;
        }

        public static bool operator != (Vector2Int a, Vector2Int b)
        {
            if (a.GetX != b.GetX || a.GetY != b.GetY)
                return true;

            return false;
        }
    }
}

File: C++/ASCII Snake/ASCII Snake/Vector2Int.cpp
#include "Vector2Int.h"

Vector2Int::Vector2Int(int x, int y)
{
	this->x = x;
	this->y = y;
}

Vector2Int::Vector2Int()
{
	x = 0;
	y = 0;
}

Vector2Int::~Vector2Int()
{
}

Vector2Int Vector2Int::add(Vector2Int other)
{
	int newX = this->x + other.x;
	int newY = this->y + other.y;

	return Vector2Int(newX, newY);
}

Vector2Int Vector2Int::subtract(Vector2Int other)
{
	int newX = this->x - other.x;
	int newY = this->y - other.y;

	return Vector2Int(newX, newY);
}

bool Vector2Int::compare(Vector2Int other)
{
	if (x == other.x && y == other.y) {
		return true;
	}

	return false;
}

Vector2Int Vector2Int::operator+(Vector2Int const& other)
{
	int newX = x + other.x;
	int newY = y + other.y;

	return Vector2Int(newX, newY);
}

bool Vector2Int::operator==(Vector2Int const& other)
{
	if (x == other.x && y == other.y) {
		return true;
	}

	return false;
}

bool Vector2Int::operator!=(Vector2Int const& other)
{
	if (x != other.x || y != other.y) {
		return true;
	}

	return false;
}

Vector2Int Vector2Int::zero()
{
	return Vector2Int(0, 0);
}
